### PR TITLE
refactoring ODEq and SDEq utils.jl to DEDiffTools

### PR DESCRIPTION
**Pull request by miguelraz**
pew pew!

### CODE REVIEW

File: src/misc_utils.jl [hunk 1]
```diff
@@ -8,27 +8,27 @@ Base.@pure function DiffCache{chunk_size}(T, length, ::Type{Val{chunk_size}})
 end
 
 Base.@pure DiffCache(u::AbstractArray) = DiffCache(eltype(u),length(u),Val{ForwardDiff.pickchunksize(length(u))})
```
Comment (ChrisRackauckas) [inline src/misc_utils.jl]: Just delete this file

File: src/StochasticDiffEq.jl [hunk 2]
```diff
@@ -20,7 +20,7 @@ module StochasticDiffEq
 
   import DiffEqBase: resize!,deleteat!,addat!,full_cache,user_cache, u_cache,du_cache,
                      resize_non_user_cache!,deleteat_non_user_cache!,addat_non_user_cache!,
-                     terminate!,get_du, get_dt,get_proposed_dt,
+                     terminate!,DiffEqDiffTools.get_du, get_dt,get_proposed_dt,
```
Comment (ChrisRackauckas) [inline src/StochasticDiffEq.jl]: Don't shadow here. This is a "different `get_du`" which is part of the integrator interface.

File: src/misc_utils.jl (hunk 1 above)
Comment (miguelraz) [inline src/misc_utils.jl]: Deleted

File: src/StochasticDiffEq.jl (hunk 2 above)
Comment (miguelraz) [inline src/StochasticDiffEq.jl]: Deleted.

**Comment by ChrisRackauckas** [comment]
You should remove DiffBase, ForwardDiff, and NLsolve from the REQUIRE file.

**Comment by miguelraz** [comment]
Removed.

**Comment by ChrisRackauckas** [comment]
Supersceded by other changes.

**Pull request closed without merging**